### DISCUSSION TITLE
Tandoor showes "You are offline"

### DISCUSSION THREAD

**Discussion by M1scer** [TandoorRecipes/recipes#2695] [Q&A]
I have installed tandoor via docker and now while it is up and running tandoor shows the "you are offline" message. But i have no clue how to fix that.
![grafik](https://github.com/TandoorRecipes/recipes/assets/93549967/490c9b74-f330-44ec-8979-bcb77525ad88)

While checking the logs i have not found anything helpful.

I am using traefik as reverse proxy. Also i had changed all volumes to bindmounts and copied the nginx config manually to the bindmound (and adjusted it according to my container names)

My compose file looks like this:
<details>
<summary>docker-compose.yml</summary>
<pre>
    tandoorDb:
        container_name: tandoorDb
        hostname: tandoorDb
        restart: always
        image: postgres:15-alpine
        volumes:
          - ./tandoor/db/data:/var/lib/postgresql/data
        env_file:
          - ./tandoor/.env
        networks:
          - traefik
        logging:
          driver: "json-file"
          options:
            max-size: "200k"
            max-file: "5"

    tandoorApp:
        container_name: tandoorApp
        hostname: tandoorApp
        restart: always
        image: vabene1111/recipes
        env_file:
          - ./tandoor/.env
        volumes:
          - ./tandoor/staticfiles:/opt/recipes/staticfiles
          - ./tandoor/nginx_config:/opt/recipes/nginx/conf.d
          - ./tandoor/mediafiles:/media:ro
        depends_on:
          - tandoorDb
        networks:
          - traefik
        logging:
          driver: "json-file"
          options:
            max-size: "200k"
            max-file: "5"

    tandoorWebserver:
        container_name: tandoorWebserver
        hostname: tandoorWebserver
        image: nginx:mainline-alpine
        restart: always
        env_file:
          - ./tandoor/.env
        volumes:
          - ./tandoor/nginx_config:/etc/nginx/conf.d:ro
          - ./tandoor/staticfiles:/static:ro
          - ./tandoor/mediafiles:/media:ro
        labels: # traefik example labels
          - "traefik.http.routers.tandoor.entrypoints=websecure"
          - "traefik.http.routers.tandoor.rule=Host(`sub.domain.de`)"
          - "traefik.http.services.tandoor.loadbalancer.server.port=80"
          - "traefik.http.routers.tandoor.tls=true"
          - "traefik.http.routers.tandoor.tls.certresolver=myresolver"
          #- "traefik.http.middlewares.tandoor.headers.sslProxyHeaders.X-Forwarded-Proto=https"
        depends_on:
          - tandoorApp
        networks:
          - traefik
        logging:
          driver: "json-file"
          options:
            max-size: "200k"
            max-file: "5"
</pre>
</details>

Log files:
<details>
<summary>Database Container</summary>
<pre>
PostgreSQL init process complete; ready for start up.
2023-10-23 08:47:14.095 UTC [1] LOG:  starting PostgreSQL 15.4 on aarch64-unknown-linux-musl, compiled by gcc (Alpine 12.2.1_git20220924-r10) 12.2.1 20220924, 64-bit
2023-10-23 08:47:14.095 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2023-10-23 08:47:14.095 UTC [1] LOG:  listening on IPv6 address "::", port 5432
2023-10-23 08:47:14.106 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2023-10-23 08:47:14.117 UTC [53] LOG:  database system was shut down at 2023-10-23 08:47:13 UTC
2023-10-23 08:47:14.130 UTC [1] LOG:  database system is ready to accept connections
2023-10-23 08:52:14.208 UTC [51] LOG:  checkpoint starting: time
2023-10-23 08:53:16.920 UTC [51] LOG:  checkpoint complete: wrote 624 buffers (3.8%); 1 WAL file(s) added, 0 removed, 0 recycled; write=62.346 s, sync=0.150 s, total=62.712 s; sync files=815, longest=0.004 s, average=0.001 s; distance=4939 kB, estimate=4939 kB
</pre>
</details>

<details>
<summary>Recipes Container</summary>
<pre>
  Applying sites.0001_initial... OK
  Applying sites.0002_alter_domain_unique... OK
  Applying socialaccount.0001_initial... OK
  Applying socialaccount.0002_token_max_lengths... OK
  Applying socialaccount.0003_extra_data_default_dict... OK
Generating static files
js-reverse file written to /opt/recipes/cookbook/static/django_js_reverse
644 static files copied to '/opt/recipes/staticfiles', 1688 post-processed.
Done
chmod: /opt/recipes/mediafiles: No such file or directory
[2023-10-23 08:48:56 +0000] [1] [INFO] Starting gunicorn 20.1.0
[2023-10-23 08:48:56 +0000] [1] [INFO] Listening at: http://0.0.0.0:8080 (1)
[2023-10-23 08:48:56 +0000] [1] [INFO] Using worker: gthread
[2023-10-23 08:48:56 +0000] [14] [INFO] Booting worker with pid: 14
[2023-10-23 08:48:56 +0000] [15] [INFO] Booting worker with pid: 15
[2023-10-23 08:48:56 +0000] [16] [INFO] Booting worker with pid: 16
</pre>
</details>

<details>
<summary>nginx Container</summary>
<pre>
/docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
/docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
/docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
10-listen-on-ipv6-by-default.sh: info: /etc/nginx/conf.d/default.conf is not a file or does not exist
/docker-entrypoint.sh: Sourcing /docker-entrypoint.d/15-local-resolvers.envsh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
/docker-entrypoint.sh: Configuration complete; ready for start up
2023/10/23 08:47:10 [notice] 1#1: using the "epoll" event method
2023/10/23 08:47:10 [notice] 1#1: nginx/1.25.2
2023/10/23 08:47:10 [notice] 1#1: built by gcc 12.2.1 20220924 (Alpine 12.2.1_git20220924-r10) 
2023/10/23 08:47:10 [notice] 1#1: OS: Linux 6.1.21-v8+
2023/10/23 08:47:10 [notice] 1#1: getrlimit(RLIMIT_NOFILE): 1048576:1048576
2023/10/23 08:47:10 [notice] 1#1: start worker processes
2023/10/23 08:47:10 [notice] 1#1: start worker process 20
2023/10/23 08:47:10 [notice] 1#1: start worker process 21
2023/10/23 08:47:10 [notice] 1#1: start worker process 22
2023/10/23 08:47:10 [notice] 1#1: start worker process 23
</pre>
</details>

**Comment by smilerz** [comment]
Most likely you are not setting headers correctly.  There are some examples of traefik configurations in issues can search for.  
If you still can't find the solution, please share the list of headers under the DEBUG listed in /system

**Reply by chetwisniewski** [reply]
I'm not getting this as well, but only when accessing it from Firefox on Android. If I use Chrome it works, and the main pages and search work, it is only when clicking to open an actual recipe it always shows the offline message. Copy and paste the link from Firefox to Chrome and no-problemo.
